lib/sla.js

```diff
@@ -34,62 +38,133 @@ function designateMixingBridge(client) {
+ * Originates a channel to a specified endpoint and places it in the Stasis app.
  * @param {Object} client - Object that contains information from the ARI
  *   connection.
+ * @param {Object} inbound- the inbound channel
  * @param {Object} bridge - Object that defines the bridge to be passed to
  *   addChannelsToBridge().
  * @return {Q} Q promise object.
  */
-function createChannel(client, bridge) {
+function originateChannel(client, inbound, bridge) {
   var defer = Q.defer();
-  var channel = client.Channel();
-  var originate = Q.denodeify(channel.originate.bind(channel));
-  originate({endpoint: 'SIP/phone', app: 'sla'})
+  var dialed = client.Channel();
+  var originate = Q.denodeify(dialed.originate.bind(dialed));
+  originate({endpoint: 'SIP/100@tcambron', app: 'sla', appArgs: 'dialed'})
     .catch(function (err) {
       defer.reject(err);
+      
     });
-  channel.once('StasisStart', function(event, channel) {
-    defer.resolve(channel);
+  dialed.once('StasisStart', function(event, dialed) {
+    var answer = Q.denodeify(dialed.answer.bind(dialed));
+    answer();
+    defer.resolve(dialed);
+  });
+  dialed.once('ChannelDestroyed', function(event, dialed) {
+    defer.resolve(null);
+  });
+  inbound.once('ChannelHangupRequest', function(event, inbound) {
+    var hangup = Q.denodeify(dialed.hangup.bind(dialed));
+    hangup();
+    defer.resolve(null);
   });
   return defer.promise;
 }
 
+/** Simple utility function for checking if the bridgeName is numerical
+ * @param {String} bridgeName - the name of the bridge to be created/used
+ * @return {boolean} whether or not the bridgeName is numerical
+ */
+function checkIfNum (string) {
+  if(!isNaN(parseInt(string))) {
+    return true;
+  }
+  else {
+    return false;
+  }
+}
+
 /**
  * Receives initial input and begins the application.
  * @param {Object} client - Client received from app.js.
  * @return {Q} Q promise object.
  */
-module.exports = function(client) {
-  return designateMixingBridge(client)
-    .then(function (bridge) {
-      return createChannel(client, bridge)
-        .then(function (channel) {
-          return addChannelsToBridge(channel, bridge);
-        });
-    });
+module.exports = function(client, inbound, bridgeName) {
+  //This specifies what bridge number to use for SLA
+  if(checkIfNum(bridgeName)) {
+    var answer = Q.denodeify(inbound.answer.bind(inbound));
+    return answer()
+      .then(function() {
+        return designateMixingBridge(client, bridgeName);
+      }).then(function (bridge) {
+        return originateChannel(client, inbound, bridge)
+          .then(function (dialed) {
+            if (dialed) {
+              return addChannelsToBridge(inbound, dialed, bridge);
+            }
+            else {
+              var hangup = Q.denodeify(inbound.hangup.bind(inbound));
+              hangup();
+              return 'Channel rejected or caller hung up';
```

Comment:
Returning a string for this is confusing. You should return a typed Error object. The caller can check the type to determine what to do with it. In other words, the program doesn't have to blow up, but returning a status as a string is error prone.